File: tournaments/sumOfDivisors/sumOfDivisors.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub sumOfDivisors {
    my ($n) = @_;

    my $result = 0;
    my $c = 1;

    while ($n >= $c) {
        if (!($n % $c)) {
            $result += $c;
        }
        ++$c;
    }

    return $result;
}

File: tournaments/sumOfDivisors/sumOfDivisors.js
function sumOfDivisors(n) {
    let result = 1;

    for (let i = 2; 1 < n; ++i) {
        i * i > n ? i = n : 0;
        let b = 0;
        while (n % i < 1) {
            let c  = result * i;
            result += c - b;
            b  = c;
            n /= i;
        }
    }

    return result;
}
